### PR TITLE
Bindings

### PR DESCRIPTION
**Pull request by flyingmachine**
Let's you specify bindings that will apply across a query term's tree. If you have the query term

```clojure
[::chapter {[::author :id] :a1}]
```

Then, for that term, if any entities reference an author's id then they'll reference `:a1`. You can apply these bindings to multiple query terms with

```clojure
(sm/bind-relations [[::site-tag :id] :st1
                    [::site-foo :id] :sf1]
  [::site-user]
  [::site-user])
```

### CODE REVIEW

**Comment by khiemlam** [comment]
Taking a look

File: test/reifyhealth/specmonstah/core_test.clj [hunk 1]
```diff
@@ -214,6 +215,113 @@
...
+(deftest bind-term-relations
+  (testing "binds entire tree when ref key is a vector"
+    ;; test helper renames :site-tag434154 to :site-tag-0
+    (is (= (->> (#'sm/bind-term-relations binding-template-relations [::site-user {:site-id :s1
+                                                                                   [::site-foo :id] :sf1}])
+                (rename-generated-keys [:site-tag]))
+           [::site-user {:site-id :s1
+                         :site-user-tag-id [:site-tag-0 {:site-id :s1
+                                                         :site-foo-id :sf1} {}]} nil]))
+    (is (= (->> (#'sm/bind-term-relations binding-template-relations [::site-user {[::site-tag :id] :st1
```
Comment (khiemlam) [inline test/reifyhealth/specmonstah/core_test.clj]: Minor: the expected value and actual value are swapped for all the assertions.

File: src/reifyhealth/specmonstah/core.clj [hunk 2]
```diff
@@ -161,6 +161,73 @@
...
+  "Ensures that relations are consistent across a query term's tree"
+  ([relations query-term]
+   (let [bindings (query-term-bindings query-term)]
+     (if (empty? bindings)
+       query-term
+       (bind-term-relations
+         relations
+         (remove-query-term-bindings query-term)
+         bindings))))
+  ([relations [ent-type refs attrs query-ref-name generated?] bindings]
```
Comment (flyingmachine) [inline src/reifyhealth/specmonstah/core.clj]: It's from https://github.com/reifyhealth/specmonstah/pull/8/files#diff-1540ea4a039eb92498e5b62a633583e7R242 though I honestly can't remember why it's necessary

**Comment by flyingmachine** [comment]
I've updated this with, hopefully, much better documentation :)

**Pull request merged**